src/checkers/inference/InferenceVisitor.java

```diff
@@ -489,13 +500,13 @@ protected void checkTypeArguments(Tree toptree,
      * @param extraArgs arguments to the error message key, before "found" and "expected" types
      */
     @Override
-    protected void commonAssignmentCheck(
+    protected boolean commonAssignmentCheck(
             Tree varTree,
             ExpressionTree valueExp,
             @CompilerMessageKey String errorKey,
             Object... extraArgs) {
         if (!validateTypeOf(varTree)) {
-            return;
+            return true;
```

Comment:
Why `true` if the tree isn't valid?

Reply:
This is similar logic in CF https://github.com/eisop/checker-framework/blob/43dcafd65db98173a02ee3e28193e143b0033ebc/framework/src/main/java/org/checkerframework/common/basetype/BaseTypeVisitor.java#L3208.

Reply:
That seems wrong over there, too, doesn't it? Let's clean this up in separate PRs.